CleverTap/Runtime/Native/UnityNativeWrapper/UnityNativeEventManager.cs

```diff
@@ -424,9 +458,108 @@ internal void FetchVariables()
 
         #endregion
 
+        #region AppInbox
+
+        internal void InitializeAppInbox()
+        {
+            UnityNativeAppInboxPersistence.Initialize(_preferenceManager);
+        }
+
+        internal void OnAppInboxMessage(List<object> inboxMessages)
+        {
+            Dictionary<string, object> message;
+            string messageId;
+            string serializedMessage;
+
+            for (int i = 0, count = inboxMessages.Count; i < count; i++)
+            {
+                message = inboxMessages[i] as Dictionary<string, object>;
+                messageId = message["_id"].ToString();
+                serializedMessage = Json.Serialize(message);
+
+                if (UnityNativeAppInboxPersistence.IsUpdatedOrNewMessage(messageId, serializedMessage))
+                {
+                    UnityNativeAppInboxPersistence.SaveMessage(messageId, serializedMessage);
+                }
+            }
+        }
```

Comment:
When messages are updated, it is also checked for any expired messages that need to be deleted.
Please check mobile SDKs and implement accordingly.

Reply:
Please check the comment above.